generate/templates/manual/revwalk/file_history_walk.cc

```diff
@@ -0,0 +1,270 @@
+NAN_METHOD(GitRevwalk::FileHistoryWalk)
+{
+  if (info.Length() == 0 || !info[0]->IsString()) {
+    return Nan::ThrowError("File path to get the history is required.");
+  }
+
+  if (info.Length() == 1 || !info[1]->IsNumber()) {
+    return Nan::ThrowError("Max count is required and must be a number.");
+  }
+
+  if (info.Length() == 2 || !info[2]->IsFunction()) {
+    return Nan::ThrowError("Callback is required and must be a Function.");
+  }
+
+  FileHistoryWalkBaton* baton = new FileHistoryWalkBaton;
+
+  baton->error_code = GIT_OK;
+  baton->error = NULL;
+  String::Utf8Value from_js_file_path(info[0]->ToString());
+  baton->file_path = strdup(*from_js_file_path);
+  baton->max_count = (unsigned int)info[1]->ToNumber()->Value();
+  baton->out = new std::vector< std::pair<git_commit *, std::pair<char *, git_delta_t> > *>;
+  baton->out->reserve(baton->max_count);
+  baton->walk = Nan::ObjectWrap::Unwrap<GitRevwalk>(info.This())->GetValue();
+
+  Nan::Callback *callback = new Nan::Callback(Local<Function>::Cast(info[2]));
+  FileHistoryWalkWorker *worker = new FileHistoryWalkWorker(baton, callback);
+  worker->SaveToPersistent("fileHistoryWalk", info.This());
+
+  Nan::AsyncQueueWorker(worker);
+  return;
+}
+
+void GitRevwalk::FileHistoryWalkWorker::Execute()
+{
+  git_repository *repo = git_revwalk_repository(baton->walk);
+  git_oid *nextOid = (git_oid *)malloc(sizeof(git_oid));
+  giterr_clear();
+  for (
+    unsigned int i = 0;
+    i < baton->max_count && (baton->error_code = git_revwalk_next(nextOid, baton->walk)) == GIT_OK;
+    ++i
+  ) {
+    // check if this commit has the file
+    git_commit *nextCommit;
+
+    if ((baton->error_code = git_commit_lookup(&nextCommit, repo, nextOid)) != GIT_OK) {
+      break;
+    }
+
+    git_tree *thisTree, *parentTree;
+    if ((baton->error_code = git_commit_tree(&thisTree, nextCommit)) != GIT_OK) {
+      git_commit_free(nextCommit);
+      break;
+    }
+
+    git_diff *diffs;
```

Comment:
oh cool! I guess since it's explained there they don't have to explain it in the function documentation for tree to tree.